lib/cppcheck.cpp

```diff
@@ -930,6 +930,26 @@ unsigned int CppCheck::checkFile(const std::string& filename, const std::string
                 // #error etc during preprocessing
                 configurationError.push_back((mCurrentConfig.empty() ? "\'\'" : mCurrentConfig) + " : [" + o.location.file() + ':' + std::to_string(o.location.line) + "] " + o.msg);
                 --checkCount; // don't count invalid configurations
+
+                if (!hasValidConfig && currCfg == *configurations.rbegin()) {
+                    // If there is no valid configuration then report error..
+                    mExitCode = 1;
```

Comment:
We should not set this arbitrarily. It is implicitly done via `Severity::error` and prevents the user from suppressing `preprocessorErrorDirective` successfully.

Reply:
thanks